src/components/MultipleAvatars.js

```diff
@@ -107,48 +108,61 @@ const MultipleAvatars = (props) => {
             {props.shouldStackHorizontally ? (
                 <>
                     {_.map([...props.icons].splice(0, 4), (icon, index) => (
-                        <View
-                            key={`stackedAvatars-${index}`}
-                            style={[
-                                styles.justifyContentCenter,
-                                styles.alignItemsCenter,
-                                StyleUtils.getHorizontalStackedAvatarBorderStyle(props.isHovered, props.isPressed, props.isInReportAction),
-                                StyleUtils.getHorizontalStackedAvatarStyle(index, overlapSize, oneAvatarBorderWidth, oneAvatarSize.width),
-                                icon.type === CONST.ICON_TYPE_WORKSPACE ? StyleUtils.getAvatarBorderRadius(props.size, icon.type) : {},
-                            ]}
+                        <Tooltip
+                            text={props.shouldShowTooltip ? icon.name : ''}
```

Comment:
```suggestion
                            text={tooltipTexts[index]}
```
Let's not repeat the `shouldShowTooltip` logic.

Reply:
This is within the for loop, we need all the data from the icons object here, not just the name

Reply:
I'm not sure I see the problem here. Can you please elaborate?
We will loop over the first 4 icons (0,1,2,3). The index here will also be (0,1,2,3) so we can just use that.

___

If we used splice(x, y) where x is not 0 then that would be a problem since we will get icons (x+1, x+2, ..., x+(y-1)) where the index values will always start from 0 (0,1,2,3, ... y).

Reply:
`tooltipTexts` is a list of just the names -> `['test@email.com', 'test2@email.com', ...]` whereas here we are looping over the icon objects which looks like 
```
[
  { 
      name: 'test@email.com', 
      source: ...,
      type: ...,
   },
  { 
      name: 'test2@email.com', 
      source: ...,
      type: ...,
   },
]

```

Reply:
Yes but inside this specific loop, we also reference the source and type keys

```
... 
<Avatar
         source={icon.source || props.fallbackIcon}
         fill={themeColors.iconSuccessFill}
         size={props.size}
         name={icon.name}
         type={icon.type}
/>

```

Reply:
okay, pushed!

Reply:
Yeah we can go either way. Using icon would work well too. Just wanted to get rid of the extra shouldShowTooltip check. But definitely not a blocker.